src/svc_vc.c

```diff
@@ -741,6 +761,119 @@ svc_vc_recv(SVCXPRT *xprt)
 		}
 
 		xd->sx_fbtbc = (int32_t)ntohl((long)xd->sx_fbtbc);
+
+		__warnx(TIRPC_DEBUG_FLAG_SVC_VC,
+			"sx_fbtbc = %08x", (int)xd->sx_fbtbc);
+
+		if (xd->sx_fbtbc == 0x0d0a0d0a) {
+			/* HA Proxy V2? */
+			uint32_t rest[2];
+			struct {
+			uint8_t ver_cmd;  /* protocol version and command */
+			uint8_t fam;      /* protocol family and address */
+			uint16_t len;     /* number of following bytes part of
+			                     the header */
+			} s;
+			union proxy_addr *pa;
+
+			rlen = recv(xprt->xp_fd, rest, sizeof(rest),
+				    MSG_WAITALL);
+			if (rlen != sizeof(rest)) {
+				__warnx(TIRPC_DEBUG_FLAG_ERROR,
+					"%s: %p fd %d proxy header failed rest rlen = %z (will set dead)",
+				__func__, xprt, xprt->xp_fd, rlen);
+				SVC_DESTROY(xprt);
+				return SVC_STAT(xprt);
+			}
+			rest[0] = ntohl(rest[0]);
+			rest[1] = ntohl(rest[1]);
+			if (rest[0] != 0x000d0a51 ||
+			    rest[1] != 0x5549540a) {
+				__warnx(TIRPC_DEBUG_FLAG_ERROR,
+					"%s: %p fd %d proxy header failed rest1=%08x rest2=%08x (will set dead)",
+				__func__, xprt, xprt->xp_fd, (int) rest[1], (int) rest[2]);
+				SVC_DESTROY(xprt);
+			}
+			rlen = recv(xprt->xp_fd, &s, sizeof(s),
+				    MSG_WAITALL);
+			if (rlen != sizeof(s)) {
+				__warnx(TIRPC_DEBUG_FLAG_ERROR,
+					"%s: %p fd %d proxy header failed header rlen = %z (will set dead)",
+				__func__, xprt, xprt->xp_fd, rlen);
+				SVC_DESTROY(xprt);
+				return SVC_STAT(xprt);
+			}
+			s.len = ntohs(s.len);
+			pa = mem_zalloc(s.len);
+			rlen = recv(xprt->xp_fd, pa, s.len, MSG_WAITALL);
+			if (rlen != s.len) {
+				__warnx(TIRPC_DEBUG_FLAG_ERROR,
+					"%s: %p fd %d proxy header rest len failed header rlen = %z (will set dead)",
+				__func__, xprt, xprt->xp_fd, rlen);
+				SVC_DESTROY(xprt);
+				return SVC_STAT(xprt);
+			}
+			if (s.ver_cmd == 0x21) {
```

Comment:
yeah, would be preferable to not have 0x21 and similar be magic numbers

Reply:
(define it somewhere)

Reply:
Again haproxy.h